tests/unittests/Accelerate/vImageTest.mm

```diff
@@ -444,6 +450,227 @@ static void vImageFillBufferWithRandomDataF(vImage_Buffer* buffer) {
+
+    result = vImageBuffer_Init(&planeBuffer[0], imageBuffer8888.height, imageBuffer8888.width, format.bitsPerComponent, 0);
+    ASSERT_TRUE(result == kvImageNoError);
+    result = vImageBuffer_Init(&planeBuffer[1], planeBuffer[0].height, planeBuffer[0].width, format.bitsPerComponent, 0);
+    ASSERT_TRUE(result == kvImageNoError);
+    result = vImageBuffer_Init(&planeBuffer[2], planeBuffer[0].height, planeBuffer[0].width, format.bitsPerComponent, 0);
+    ASSERT_TRUE(result == kvImageNoError);
+    result = vImageBuffer_Init(&planeBuffer[3], planeBuffer[0].height, planeBuffer[0].width, format.bitsPerComponent, 0);
+    ASSERT_TRUE(result == kvImageNoError);
+
+    // Note: Although the function calls for ARGB input, a different input format can be used if the output planes are swizzled
+    if (indexA == 0) {
+        if (indexR == 1) {
+            // XRGB or ARGB
+            result = vImageConvert_ARGB8888toPlanar8(&imageBuffer8888,
+                                                     &planeBuffer[indexA],
+                                                     &planeBuffer[indexR],
+                                                     &planeBuffer[indexG],
+                                                     &planeBuffer[indexB],
+                                                     0);
+        } else {
+            // XBGR or ABGR
+            result = vImageConvert_ARGB8888toPlanar8(&imageBuffer8888,
+                                                     &planeBuffer[indexA],
+                                                     &planeBuffer[indexB],
+                                                     &planeBuffer[indexG],
+                                                     &planeBuffer[indexR],
+                                                     0);
+        }
+    } else if (indexA == 3) {
+        if (indexR == 0) {
+            // RGBX or RGBA
+            result = vImageConvert_ARGB8888toPlanar8(&imageBuffer8888,
+                                                     &planeBuffer[indexR],
+                                                     &planeBuffer[indexG],
+                                                     &planeBuffer[indexB],
+                                                     &planeBuffer[indexA],
+                                                     0);
+        } else {
+            // BGRX or BGRA
+            result = vImageConvert_ARGB8888toPlanar8(&imageBuffer8888,
+                                                     &planeBuffer[indexB],
+                                                     &planeBuffer[indexG],
+                                                     &planeBuffer[indexR],
+                                                     &planeBuffer[indexA],
+                                                     0);
+        }
+    }
+
+    ASSERT_TRUE(result == kvImageNoError);
+    const uint32_t height = (uint32_t)imageBuffer8888.height;
+
+    unsigned char* colorData;
```

Comment:
>colorData [](start = 19, length = 9)

nit: Prefer declaring and initializing these variables at the same time unless it is unavoidable.